### PR TITLE
Create an account on new order (#214)

### PR DESCRIPTION
**Pull request by rus-daniel**

### CODE REVIEW

File: app/code/community/Oyst/OneClick/Model/Magento/Quote.php [hunk 1]
```diff
@@ -93,10 +96,58 @@ private function initializeQuote($storeId = null)
...
+    {
+        $customer = Mage::getModel('customer/customer');
+        $store = Mage::app()->getStore();
+
+        $customer->setWebsiteId($this->websiteId)
+            ->setStore($store)
+            ->setFirstname($firstname)
+            ->setLastName($lastname)
+            ->setEmail($email)
+            ->setPassword($this->generatePassword());
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/Magento/Quote.php]: use the Magento ```generatePassword()```
https://github.com/OpenMage/magento-mirror/blob/f9188bdc4df444a2c2b34a401375c5d12fbbfd2e/app/code/core/Mage/Customer/Model/Customer.php#L420

File: app/code/community/Oyst/OneClick/etc/system.xml [hunk 2]
```diff
@@ -572,6 +572,26 @@
...
+                    <sort_order>60</sort_order>
+                    <show_in_default>1</show_in_default>
+                    <show_in_website>1</show_in_website>
+                    <show_in_store>1</show_in_store>
+                    <fields>
+                        <create translate="label">
+                            <label>Create account for new order</label>
+                            <frontend_type>select</frontend_type>
+                            <source_model>adminhtml/system_config_source_yesno</source_model>
+                            <config_path>oyst/oneclick/create_account</config_path>
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/etc/system.xml]: Could you add the default value in the ```config.xml```

File: app/code/community/Oyst/OneClick/Model/Magento/Quote.php [hunk 3]
```diff
@@ -93,10 +96,39 @@ private function initializeQuote($storeId = null)
         Mage::getSingleton('checkout/session')->replaceQuote($this->quote);
     }
 
+    /**
+     * Create new customer.
+     *
+     * @param string $firstname
+     * @param string $lastname
+     * @param  string $email
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/Magento/Quote.php]: use less space

File: app/code/community/Oyst/OneClick/etc/system.xml [hunk 4]
```diff
@@ -572,6 +572,26 @@
                     </fields>
                 </reset>
 
+                <new_account translate="label">
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/etc/system.xml]: change ```new_account``` to ```new_customer_account```

File: app/code/community/Oyst/OneClick/etc/config.xml [hunk 5]
```diff
@@ -201,6 +201,7 @@
                 <product_page_buttons_wrapper_class>add-to-cart-buttons</product_page_buttons_wrapper_class>
                 <button_rounded>1</button_rounded>
                 <button_smart>1</button_smart>
+                <create_account>1</create_account>
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/etc/config.xml]: change ```new_account``` to ```new_customer_account```

File: app/code/community/Oyst/OneClick/etc/system.xml (hunk 2 above)
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/etc/system.xml]: change ```new_account``` to ```new_customer_account```

File: app/code/community/Oyst/OneClick/etc/system.xml [hunk 6]
```diff
@@ -572,6 +572,26 @@
...
+                    <label>Create account for new order</label>
```
Comment (StefBrito) [inline app/code/community/Oyst/OneClick/etc/system.xml]: "Create a customer account for a new order"

French "Création d'un compte utilisateur lors d'une nouvelle commande"

File: app/code/community/Oyst/OneClick/etc/system.xml [hunk 7]
```diff
@@ -572,6 +572,26 @@
...
+                    <frontend_type>text</frontend_type>
+                    <sort_order>60</sort_order>
+                    <show_in_default>1</show_in_default>
+                    <show_in_website>1</show_in_website>
+                    <show_in_store>1</show_in_store>
+                    <fields>
+                        <create translate="label">
+                            <label>Create account for new order</label>
```
Comment (StefBrito) [inline app/code/community/Oyst/OneClick/etc/system.xml]: "Create a customer account for a new order"

French "Création d'un compte utilisateur lors d'une nouvelle commande"

**Pull request merged**